### ISSUE TITLE
Buggy cursor graphic when capturing the game RuneScape

### ISSUE THREAD

**Issue by windwakr** [obsproject/obs-studio#3472]
## Platform

Operating system and version: Windows 7 64bit
OBS Studio version: 25.0.08

## Expected Behavior

I expect the cursor to appear correctly in the recorded video.

## Current Behavior

The cursor appears buggy in the recorded video.

## Steps to Reproduce

1. Download the client for RuneScape and start it up. The game dynamically downloads its cache, so the download here should be small. https://www.runescape.com/download
2. You do not need to make an account or log in, as the issue is apparent at the log in screen.
2. Click the gear in the top right of the login window and make sure custom cursors are enabled.
3. Capture the game and you will see the buggy cursor in the recorded video.

## Additional information

This problem has been around for a long time. You can find videos on YouTube from years ago with this buggy cursor.
![Untitled](https://user-images.githubusercontent.com/284886/93156695-a763ef00-f6d6-11ea-8530-0fdd2f92c880.png)

The game does have the option to disable custom cursors, but that's less than desirable.

**Comment by Fenrirthviti** [comment]
Can you confirm if this happens on supported operating system, such as Windows 10?

**Comment by WizardCM** [comment]
I can confirm this occurs on 25.0.8 on Windows 10 2004.

![image](https://user-images.githubusercontent.com/941350/93162931-2f8ad880-f759-11ea-8fd5-08aada206e12.png)

We don't do anything special with cursors, but we're currently tracking cursor issues in #2604

**Comment by windwakr** [comment]
Just speculating here, as I don't have a build environment set up to test things. I've been looking through the code and I believe "plugins/win-capture/cursor-capture.c" is responsible for the cursor, correct? The function copy_from_color tests the main bitmap for being 32 bits per pixel, but not the mask. Is it possible that the mask in this case is less than 32bpp? I think that might explain the pattern we're seeing on the cursor graphic.

**Comment by windwakr** [comment]
I believe I've found the source of this problem.
In [plugins/win-capture/cursor-capture.c](https://github.com/obsproject/obs-studio/blob/531120da9956d5ae2ba9eff1ae4fb7f79dd9249c/plugins/win-capture/cursor-capture.c) the function bit_to_alpha is to blame. This cursor is 13x19 pixels, but the mask is 16x19. There's padding to bmWidthBytes.

**Comment by windwakr** [comment]
In addition to one or more of the above mentioned functions needing fixing, so does get_bitmap_data. It doesn't allocate enough memory for the whole mask.

**Comment by windwakr** [comment]
Sorry for being a bit spammy, but just wanted to confirm that the previous posts are correct about the source of the issue. I was able to do some actual testing with OBS.
I created a fork at https://github.com/windwakr/obs-studio/tree/cursorfix which shows two changes I made to make this function correctly. I'm not sure if this is the best way to go about this, though.
![cursors](https://user-images.githubusercontent.com/284886/93274426-557ba180-f788-11ea-803f-41e8bb332267.png)
